services/services.go

```diff
@@ -684,70 +677,118 @@ func getIndexPageData() (*types.IndexPageData, error) {
 		return nil, fmt.Errorf("error retrieving index block data: %v", err)
 	}
 
-	blocksMap := make(map[uint64]*types.IndexPageDataBlocks)
-	for _, block := range blocks {
-		if blocksMap[block.Slot] == nil || len(block.BlockRoot) > len(blocksMap[block.Slot].BlockRoot) {
-			blocksMap[block.Slot] = block
+	// keep slots with higher blockroot length
+	slotsMap := make(map[uint64]*types.IndexPageDataBlocks)
+	for _, slot := range slots {
+		if slot.Slot > (slots[0].Slot-10) && (slotsMap[slot.Slot] == nil || len(slot.BlockRoot) > len(slotsMap[slot.Slot].BlockRoot)) {
```

Comment:
You could `minSlot := slots[0].Slot-10` before the for loop and use this here to save some time.